File: app/src/main/java/com/attendance/myproject/begory/Utilities/SavePrefs.java
package com.attendance.myproject.begory.Utilities;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

import com.google.gson.Gson;

public class SavePrefs<T> {
	private String USER_PREFS_FILE_NAME;
	private final String DATA = "DATA";
	private SharedPreferences prefs;
	private Class<?> cls;

	public SavePrefs(Context activity, Class<?> cls) {
		super();
		this.cls = cls;
		USER_PREFS_FILE_NAME = cls.getName();
		prefs = activity.getSharedPreferences(USER_PREFS_FILE_NAME,
				Context.MODE_PRIVATE);
	}

	public void save(T obj) {
		Editor editor = prefs.edit();
		String data = new Gson().toJson(obj);

		editor.putString(DATA, data);
		editor.commit();
	}

	@SuppressWarnings("unchecked")
	public T load() {
		String data = prefs.getString(DATA, "");
		return (T) new Gson().fromJson(data, cls);
	}

	public void clear() {
		prefs.edit().clear().commit();
	}

}

File: app/src/main/java/com/attendance/myproject/begory/data/source/local/prefs/SavePrefs.kt
package com.attendance.myproject.begory.data.source.local.prefs

import android.content.Context
import android.content.SharedPreferences
import com.attendance.myproject.begory.di.PreferenceInfo
import com.attendance.myproject.begory.di.module.AppModule
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject
import javax.inject.Singleton

class SavePrefs<T> @Inject constructor(@ApplicationContext activity: Context, @PreferenceInfo preferenceInfo: String, private val cls: Class<*>) {

    private val prefs: SharedPreferences = activity.getSharedPreferences(preferenceInfo,
            Context.MODE_PRIVATE)

    fun save(Pref_key:String,obj: T) {
        val editor = prefs.edit()
        val gson =GsonBuilder().serializeNulls().create();
        val data = gson.toJson(obj )
        editor.putString(Pref_key, data)
        editor.apply()
    }

    fun load(Pref_key:String): T {
        val data = prefs.getString(Pref_key,"")
        return Gson().fromJson(data, cls) as T
    }
    fun savestring(Pref_key:String,obj: String) {
        val editor = prefs.edit()
        editor.putString(Pref_key, obj)
        editor.apply()
    }

    fun loadstring(Pref_key:String): String {
        return prefs.getString(Pref_key,"")
    }
    fun clear() {
        prefs.edit().clear().apply()
    }
}